enhancements/aiops/20230216-enable-gpu-services-training.md

```diff
@@ -0,0 +1,73 @@
+# Title: 
+Enable GPU Services upon submission of initial training job for workload logs.
+
+## Summary: 
+Currently, when a user would like to enable AI services, they will go to the Opni Admin Dashboard. First, they are required to enable Logging and once that is done, they will go to the AIOps panel and check the "Enable GPU Services button". When they hit the "Save" button, the GPU services are installed within the Kubernetes cluster. This includes the workload DRAIN service, the training controller service, the GPU Controller service and CPU Inferencing service. This UX can be avoided by simply detecting the availability of a GPU within the cluster and when the user creates or updates the workload log anomaly watchlist to train a Deep Learning model, that is when these GPU services should be installed, rather than through a checkbox button.
+
+## Use case: 
+This will remove the "Enable GPU Services" check box and now will install the GPU services when the user decides to update the watchlist for the very first time with workloads. Opni GPU services will automatically come up upon the creation or update of a workload log anomaly watchlist.
+
+## Benefits: 
+* Improves the usability of Opni AIOps log anomaly
+
+
+## Impact: 
+The impact of this feature would be on several services. The AIOps gateway plugin will need to be modified to be the one which enables the GPU services when the workload watchlist is enabled for the first time. Additionally, since the update of the watchlist will be launching the GPU services, logic will have to be added to AIOps gateway plugin to update the current watchlist of workloads into Opensearch and the training controller service will need to fetch these workloads from Opensearch upon startup. Lastly, the Opni Admin Dashboard UI will also be modified to no longer include the check box.
+
+## Implementation details:
+
+There are a couple of different scenarios that will be addressed.
+
+Scenario 1: GPU is not available in central Opni cluster
+In this case, the Opni Admin Dashboard will make a request to the gpu_info endpoint within the AIOps gateway. When, it receives the response that the GPU is not available. It will still display the deployments by cluster and namespace. However, the user will not be able to update the watchlist as that button will be greyed out. Additionally, a banner will be displayed within the Opni Admin Dashboard informing the user that they need to configure a GPU on this cluster to receive insights and it will point them to a link for reference.
+
+Scenario 2: GPU is available in central Opni cluster
+
+![GPU Services Modification Diagram (1)](https://user-images.githubusercontent.com/8761010/221273618-c3a88b3e-d169-4951-b790-5b3a1c8f41f6.png)
```

Comment:
This diagram is confusing and doesn't follow conventions. Fix this
Going with the following sets things up clearer
![Untitled-2023-02-23-1130](https://user-images.githubusercontent.com/6551215/221645821-a1e6eee3-8516-4267-9fda-59acc6e607b1.png)